doc/scripts/gen_state_diagram.py

```diff
@@ -140,14 +140,14 @@ def get_event(self, line):
                                  line):
                 if i.group(1) not in self.edges.keys():
                     self.edges[i.group(1)] = []
-                if len(self.context) is 0:
+                if not len(self.context):
```

Comment:
instead of using `not`, i'd suggest either compare it with 0 or just put
```py
if self.context:
```
as `self.context` is a list.

Reply:
`not` will be required here to check empty list. I have removed list length calculation.